lib/security_alert_handler.rb

```diff
@@ -0,0 +1,101 @@
+class SecurityAlertHandler
+  def initialize(github, organisation, repos)
+    @github = github
+    @organisation = organisation
+    @repos = repos
+    @alerts_fetch_success = true
+    @global_security_alerts = fetch_security_alerts
+  end
+
+  def security_alerts_count
+    return nil unless @alerts_fetch_success
+
+    @global_security_alerts.length
+  end
+
+  def filter_security_alerts(repo)
+    @global_security_alerts.select { |alert| alert[:repo] == repo }
+  end
+
+  def label_for_branch(branch, title, alerts_for_repo)
+    security_label_for_branch(branch, title, alerts_for_repo)
+  end
+
+private
+
+  def fetch_security_alerts
+    alerts = []
+    success_count = 0
+    @repos.each do |repo|
+      response = @github.get("https://api.github.com/repos/#{@organisation}/#{repo}/dependabot/alerts", accept: "application/vnd.github+json", state: "open")
+      alerts.concat(response.map { |alert| parse_security_alert(alert, repo) })
+      success_count += 1
+    rescue StandardError => e
+      puts "Error fetching security alerts for repo #{repo}: #{e.message}"
+      next
+    end
+    @alerts_fetch_success = success_count.positive?
+    alerts
+  end
+
+  def parse_security_alert(alert, repo)
+    {
+      repo:,
+      package_manager: alert[:dependency][:package][:ecosystem],
+      dependency_name: alert[:dependency][:package][:name],
+      vulnerable_version_ranges: alert[:security_vulnerability][:vulnerable_version_range].split(", ").map(&:strip),
+      url: alert[:html_url],
+      severity: alert[:security_vulnerability][:severity],
+    }
+  end
+
+  def extract_versions_from_title(title)
+    match = title.match(/Bump (?<dependency_name>[\w-]+) from (?<old_version>[\d.]+) to (?<new_version>[\d.]+)/)
+    return nil unless match
+
+    {
+      old_version: match[:old_version],
+      new_version: match[:new_version],
+    }
+  end
```

Comment:
Interesting that we've solved the same problem as in https://github.com/alphagov/govuk-dependency-checker/pull/7/files#diff-74fbaa7d9eb23356d6a00d106eda732b13cf02489f86b9e813e738d7648e5670R28-R38. (Don't think there's anything we can do about it but worth noting to ponder 😁 possible we could consolidate the code in future)